### ISSUE TITLE
md5 checksum hash comparison borken between 2/3

### ISSUE THREAD

**Issue by EvanPurkhiser** [EvanPurkhiser/dots#23]
Need to look into this.

**Comment by EvanPurkhiser** [comment]
I did a poor job documenting what was actually broken here. I've just made 53c613c0fbe7351a663503533699af866690b1cd. Quite possibly either fixing this, or breaking things again.

**Comment by EvanPurkhiser** [comment]
Ok yep. hash comparison is now broken in python3.

```
Traceback (most recent call last):
  File "/home/evan/.local/bin/dots", line 609, in <module>
    config.install_tree(args.location, args.file, True)
  File "/home/evan/.local/bin/dots", line 284, in install_tree
    [f.install(to, exec_install_scripts) for f in self.files(path_filters)]
  File "/home/evan/.local/bin/dots", line 284, in <listcomp>
    [f.install(to, exec_install_scripts) for f in self.files(path_filters)]
  File "/home/evan/.local/bin/dots", line 522, in install
    compiled = hashlib.md5(compiled).digest()
TypeError: Unicode-objects must be encoded before hashing
```

**Comment by EvanPurkhiser** [comment]
Ok. Fairly sure this is now fixed by f08e714dac0e07e2c3407adac4cf15d42753cf36
